pkg/cmd/target/target.go

```diff
@@ -18,33 +17,31 @@ import (
 )
 
 // NewCmdTarget returns a new target command.
-func NewCmdTarget(f util.Factory, o *TargetOptions) *cobra.Command {
+func NewCmdTarget(f util.Factory, ioStreams util.IOStreams) *cobra.Command {
+	o := &TargetOptions{
+		Options: base.Options{
+			IOStreams: ioStreams,
+		},
+	}
 	cmd := &cobra.Command{
 		Use:   "target",
-		Short: "Set scope for next operations, e.g. \"gardenctl target garden garden_name\" to target garden with name of garden_name",
-		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
-			suggestions, err := validTargetArgsFunction(f, o, args, toComplete)
-			if err != nil {
-				fmt.Fprintln(o.IOStreams.ErrOut, err.Error())
-				return nil, cobra.ShellCompDirectiveNoFileComp
-			}
-
-			return util.FilterStringsByPrefix(toComplete, suggestions), cobra.ShellCompDirectiveNoFileComp
-		},
-		RunE: func(cmd *cobra.Command, args []string) error {
-			if err := o.Complete(f, cmd, args); err != nil {
-				return fmt.Errorf("failed to complete command options: %w", err)
-			}
+		Short: "Set scope for next operations, using subcommands or pattern",
+		Example: `# target project "my-project" of garden "my-garden"
+gardenctl target --garden my-garden --project my-project
 
-			if err := o.Validate(); err != nil {
-				return err
-			}
+# target shoot "my-shoot" of currently selected project
+gardenctl target shoot my-shoot
 
-			return runCmdTarget(f, o)
-		},
+# Target shoot control-plane using values that match a pattern defined for a specific landscape
```

Comment:
```suggestion
# Target shoot control-plane using values that match a pattern defined for a specific garden
```
as you were replacing `landscape` with garden in one of your latest commits, I guess this one also needs to be replaced

Reply:
Yes, I changed it to garden to not confuse the users. We always use the term garden, yes a garden identifies a landscape but nevertheless I thought it would be better to keep things simple.